Added Toggle GUI option to toggle on/off the Menu+Status bars

Sembiance opened the pull request:
Added a "Toggle GUI" menu item in the File menu with shortcut (F10) which will toggle on and off the visibility of the Menu Bar and Status Bar.

I do this because I run in windowed mode and want to hide these elements to as not distract from my NES gaming goodness.

By the way, thanks for making such an amazing emulator!

punesemu commented:
Hi Sembiance, Thx for the patch and for your your support. I've reviewed your patch and this is the [result](https://gist.github.com/punesemu/3a45c60b40043d4e97345987dd73b5d0) . I preferred to associate the shortcut 'ALT+G' because the F10 is too close to F11 that is the Hard Reset. I wouldn't want it to be pressed by mistake. Thanks to your patch I also fixed a bug.

Sembiance commented:
Cool, I applied the patch and was able to toggle off the bars.

One other thing I changed is that in Linux, there are "2 pixels" of extra space at the bottom of the window. If you look at this screenshot: https://i.imgur.com/fEEI7u8.png
You can see the 2 pixel high white bar going along the bottom.

It seems to be added in src/gui/qt.zip:
`(qt.screen->height() + 2) +`

Not sure why those 2 pixels are there, maybe just for extra padding between the game and the bottom status bar.

I had changed it to this:
`(qt.screen->height() + (qt.mwin->menubar->isHidden() && qt.mwin->statusbar->isHidden() ? 0 : 2)) +`

Which got rid of those 2 extra pixels when both top and bottom bars are hidden.

This is something that your original code already took into account when switching to "full screen" by subtracting 2 pixels.

Your new patch doesn't seem to do this which does leave a 2px high horizontal white bar at the bottom of the window after hiding the GUI. Not sure if that's on purpose or not :)

punesemu commented:
I've  updated the [patch](https://gist.github.com/punesemu/3a45c60b40043d4e97345987dd73b5d0) also including the possibility of execute the toggle even when the emu is in fullscreen mode and eliminating those two pixels that I had completely forgotten, I should have eliminated them already some release ago.

Sembiance commented:
Excellent! Just applied the patch and tested it out, works GREAT. No extra pixels. Thanks! I'll close this pull request and then you can merge your patch into puNES whenever you wish.

Thanks again!

RokkumanX commented:
Excellent indeed, great work @Sembiance

punesemu commented:
https://github.com/punesemu/puNES/commit/4ddde7d9bc2ce56e243d8614841d01436b2e10ec

anewuser commented:
Can this option be used with a command-line argument?

punesemu commented:
https://github.com/punesemu/puNES/commit/49aec86890a3342464f296e3f7b36ad2fbdc2ce5
